sdk/src/main/java/io/customer/sdk/repository/PreferenceRepository.kt

```diff
@@ -0,0 +1,41 @@
+package io.customer.sdk.repository
+
+import android.content.Context
+import android.content.SharedPreferences
+
+internal interface PreferenceRepository {
+    fun saveIdentifier(identifier: String)
+    fun removeIdentifier(identifier: String)
+    fun getIdentifier(): String?
+}
+
+internal class PreferenceRepositoryImpl(private val context: Context) : PreferenceRepository {
+
+    companion object {
+        private const val PREFS_NAME = "customer_io_store"
+        private const val KEY_IDENTIFIER = "identifier"
+    }
+
+    private val prefs: SharedPreferences by lazy {
+        context.applicationContext.getSharedPreferences(
+            PREFS_NAME,
```

Comment:
On iOS, we use the site id as the ID for the shared preferences store that we retrieve. This avoids the scenario when someone wants to change Customer.io workspaces in their app or they want to use multiple workspaces per app. 

Would that be better to do then having all data stored in 1 shared preferences store?

Reply:
Absolutely, I missed the case for multiple workspaces. Would update as `site ID`being key.